aws/cloudformation-templates/mparticle.yaml

```diff
@@ -0,0 +1,135 @@
+---
+AWSTemplateFormatVersion: 2010-09-09
+
+Description: >
+    This template deploys a Kinesis stream and two Lambda functions for use with the Amazon Personalize destination 
+    in the Personalize and CDP workshops, along with the execution roles required to run the Lambda and for mParticle
+    to be able to access the Lambda from the Personalize destination.
+
+Parameters:
+  ResourceBucket:
+    Type: String
+    Description: >
+      S3 bucket name where the Retail Demo Store deployment resources are staged (product images, nested CloudFormation templates, source code snapshot, 
+      notebooks, deployment Lambda code, etc).  You can substitute your own bucket here if needed.
+
+  ResourceBucketRelativePath:
+    Type: String
+    Description: >
+      Optional path in the Deployment Resources Staging bucket where the deployment resources are stored (e.g. path/path2/). 
+      Leave blank if resources are at the root of the Staging Resource Bucket. If specified, MUST end with '/'.
+
+  Uid:
+    Type: String
+    Description: >
+      Uid generated from the root template to provide unique resource names
+
+  mParticleOrgId:
+    Type: String
+    Description: >
+      The OrgID for your mParticle instance.  This will be used to enable assume role for mParticle to write to Kinesis.
+
+Resources:
+  mParticlePersonalizeEventsKinesisStream:
+    Type: AWS::Kinesis::Stream
+    Properties: 
+      Name: !Sub '${Uid}-mParticlePersonalizeEventsKinesisStream'
+      ShardCount: 1
+
+  mParticlePersonalizeLambda:
+    Type: 'AWS::Lambda::Function'
+    Properties:
+      Description: 'Handles sending events passed from mParticle to the Personalize tracker for user-item interactions.'
+      Handler: mparticle-personalize.handler
+      Role: !GetAtt 
+        - mParticlePersonalizeLambdaExecutionRole
+        - Arn
+      Code:
+        S3Bucket: !Ref ResourceBucket
+        S3Key: !Sub '${ResourceBucketRelativePath}aws-lambda/mparticle-personalize.zip'
+      Runtime: nodejs12.x
+      Timeout: 900
+      FunctionName: !Sub '${Uid}-mParticlePersonalizeLambda'
+      Environment:
+        Variables:
+          PERSONALISE_TRACKING_ID:  ''
+          PERSONALISE_CAMPAIGN_ARN: ''
+          MPARTICLE_S2S_API_KEY: ''
+          MPARTICLE_S2S_SECRET_KEY: ''
```

Comment:
Shouldn't these be looked up from SSM?

Reply:
This won't be available during the creation of the stack within CF.  What Igor and I have done is basically, within the Jupyter notebook for mParticle, to set the auto-generated values after the stack has been created into SSM which will be pulled eventually by the dependencies. I'll let Igor explain further here as he was the one who helped me build the cloud formation template. Cheers

Reply:
If these values are pulled from SSM, are the environment variables needed here? The function does not do anything with these variables. It looks like these can be deleted.

Reply:
Fixed this issue - these are not needed as they are being set exclusively via SSM - we should move to using this model to simplify the CF templates as there is a lot of error-prone variable passing required otherwise